### ISSUE TITLE
Scraper crashes with specific url pattern

### ISSUE THREAD

**Issue by samaloney** [sunpy/sunpy#4493]
### Description

Scraper has an issue handling a specific url pattern but works fine for a very similar pattern.

Works: `r'http://pub023.cs.technik.fhnw.ch/data/new/'{level}/%Y/%m/%d/{type}/solo_{level}_stix-{product}_%Y%m%d_V01\\.fits`

Breaks: `r'http://pub023.cs.technik.fhnw.ch/data/new/'{level}/%Y/%m/%d/{type}/solo_{level}_stix-{product}_%Y%m%dT%H%M%S_V01\\.fits'`
### Expected vs Actual behaviour

Scraper to parse the url pattern and return a file list. The error occur on line 194 as `date_part` is `None`. Locally it seems like including lines 194-198 in an `if date_part:` block fixes this I but haven't run the tests to see if this breaks anything else yet. Also not totally clear as to the root cause so this may not be the best solution.

https://github.com/sunpy/sunpy/blob/d2fd01d6ec5e064f1575c7788e8b1c9c4188c6b2/sunpy/util/scraper.py#L190-L200

### Steps to Reproduce

```python
from sunpy.time import TimeRange
from sunpy.util.scraper import Scraper

timerange = TimeRange('2020-05-23T00:00','2020-05-25T00:00')
pattern = 'http://pub023.cs.technik.fhnw.ch/data/new/{level}/%Y/%m/%d/{type}/solo_{level}_stix-{product}_%Y%m%dT%H%M%S_V01\\.fits'
scraper = Scraper(pattern, level='L1',type='SCI', product='.*')
files = scraper.filelist(timerange)
```
The actual error occurs when the scraper tries to extract the date from a given url using `_extractDateURL ` so this example shows the problem without needing to query a server/network.

```python
from sunpy.util.scraper import Scraper

pattern = 'http://pub023.cs.technik.fhnw.ch/data/new/{level}/%Y/%m/%d/{type}/solo_{level}_stix-{product}_%Y%m%dT%H%M%S_V01\\.fits'

scraper = Scraper(pattern, level='L1',type='SCI', product='.*')

scraper._extractDateURL('http://pub023.cs.technik.fhnw.ch/data/new/L1/2020/05/24/SCI/solo_L1_stix-sci-xray-l1-1170327568_20200524T075117_V01.fits')
```

### System Details

- SunPy Version: 2.0.1
- Astropy Version: 3.2.3
- Python Version: 3.7.0
- OS information: macOS 10.15.6

**Comment by samaloney** [comment]
After looking at this a bit more it a combination of RTFM/user error  the docs state you can't use regex and keyword substitution at the same time.

However when switching over to regex processing I found that you can't use any dots in you regex e.g. `.*` as they are replaced at the start of processing which is not documented.

https://github.com/sunpy/sunpy/blob/d2fd01d6ec5e064f1575c7788e8b1c9c4188c6b2/sunpy/util/scraper.py#L149-L155

**Comment by dstansby** [comment]
I am wondering if the solution here is to fix the code to stop it splitting the regex at dots, or to document the current behaviour?

**Comment by samaloney** [comment]
Yea its a bit odd that it takes a regex but doesn't support full python regex

**Comment by jeffreypaul15** [comment]
This issue doesn't occur when setting regex=True while creating an object of the Scraper class, it works in both cases where @samaloney has mentioned that it breaks when passing in a pattern with time apart from the date.
```python
from sunpy.util.scraper import Scraper

works = 'http://pub023.cs.technik.fhnw.ch/data/new/{level}/%Y/%m/%d/{type}/solo_{level}_stix-{product}_%Y%m%d_V01\\.fits'
scraper = Scraper(works, regex=True)
scraper._extractDateURL('http://pub023.cs.technik.fhnw.ch/data/new/L1/2020/05/24/SCI/solo_L1_stix-sci-xray-l1-1170327568_20200524T075117_V01.fits')
```
``` Output : <Time object: scale='utc' format='isot' value=2020-05-24T00:00:00.000> ```

```python
breaks = 'http://pub023.cs.technik.fhnw.ch/data/new/{level}/%Y/%m/%d/{type}/solo_{level}_stix-{product}_%Y%m%dT%H%M%S_V01\\.fits'
scraper = Scraper(breaks, regex=True)
scraper._extractDateURL('http://pub023.cs.technik.fhnw.ch/data/new/L1/2020/05/24/SCI/solo_L1_stix-sci-xray-l1-1170327568_20200524T075117_V01.fits')
```
``` <Time object: scale='utc' format='isot' value=2020-05-24T07:51:17.000> ```

Should this be mentioned in the docs?

**Comment by nabobalis** [comment]
Yeah it is worth adding a note or something similar to the docs just to warn peope.

**Comment by jeffreypaul15** [comment]
I'll make a PR regarding this but where exactly should this be mentioned?

**Comment by nabobalis** [comment]
> I am wondering if the solution here is to fix the code to stop it splitting the regex at dots, or to document the current behaviour?

The solution here is to rewrite scraper with a new design.

**Comment by nabobalis** [comment]
> I'll make a PR regarding this but where exactly should this be mentioned?

The main scrapper class is enough?

**Comment by kakirastern** [comment]
Would really side with @nabobalis this time 🤔

**Comment by wtbarnes** [comment]
I'm having trouble reproducing the reported behavior given that the URLs in the original example don't seem to exist anymore. Given this and the massive rewrite of `Scraper` in #7077 and #7277, I'm going to close this. Feel free to reopen if there is still some issue that needs to be addressed.
